ui/com/user-info.jsx

```diff
@@ -25,8 +25,8 @@ const FLAG_DROPDOWN = [
 class AutoRefreshingComponent extends React.Component {
   constructor(props) {
     super(props)
-    this.state = this.computeState()
-    this.refreshState = props => { this.setState(this.computeState(props)) }
+    this.computeState(null, this.setStateCallback.bind(this))
```

Comment:
Ah yes, I knew you could declare variables inside functions, just feels strange and slightly counter my intuition for that to be a function. I think in Ruby it would be outside. Thanks for pointing that out. 

Thanks for the clear write up about `=>` and `bind` as well, I didn't know that detail explicitly